### PR TITLE
Update joinSub docblock

### PR DESCRIPTION
**Pull request by harmenjanssen**
**Changes**

The `joinSub` docblock in the `QueryBuilder.stub` file is not (no longer) aligned with [Laravel 9](https://github.com/laravel/framework/blob/v9.7.0/src/Illuminate/Database/Query/Builder.php#L526).
The stub version actually accepts a narrower range of types than Laravel itself. 
In particular, `\Illuminate\Database\Eloquent\Builder` was not accepted, but is accepted according to Laravel.

This change fixes the docblock and thereby removes a PHPStan false negative.

**Breaking changes**

None.

### CODE REVIEW

**Comment by szepeviktor** [comment]
Hello Harmen! 👋 
Could you add a test that fails without this PR?

**Comment by harmenjanssen** [comment]
Hi Viktor! 🙂 

I've tried adding a test, by simply using `joinSub` in the feature test, and passing it an Eloquent builder. 
But I can see even before that, the type in the stub file is incorrect: 

```
Method Illuminate\\Database\\Query\\Builder::joinSub() has parameter $query with generic class Illuminate\\Database\\Eloquent\\Builder but does not specify its types: TModelClass
```

I'm not entirely sure how to fix this. I've taken the type from Laravel 9's own docblock. 
Can you give me a pointer on how to debug?

**Comment by harmenjanssen** [comment]
Yeah, I understand. I just don't know how to mark it as such in the type signature. 
I cannot add `\Illuminate\Database\Eloquent\Builder<TModelClass>` in the `@param` doc. 
It will give me the error:

```
Type Illuminate\\Database\\Query\\TModelClass in generic type Illuminate\\Database\\Eloquent\\Builder<Illuminate\\Database\\Query\\TModelClass> in PHPDoc tag @param for parameter $query is not subtype of template type TModelClass of Illuminate\\Database\\Eloquent\\Model of class Illuminate\\Database\\Eloquent\\Builder.
```

I'm not sure how to refer to the original Eloquent Builder's `TModelClass` template type...

File: stubs/QueryBuilder.stub [hunk 1]
```diff
@@ -135,7 +135,7 @@ class Builder
     /**
      * Add a subquery join clause to the query.
      *
-     * @param  \Closure|\Illuminate\Database\Query\Builder|string  $query
+     * @param  \Closure|\Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder|string  $query
```
Comment (canvural) [inline stubs/QueryBuilder.stub:138]:
```suggestion
     * @param  \Closure|\Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder<\Illuminate\Database\Eloquent\Model>|string  $query
```

Maybe you can try this

**Comment by harmenjanssen** [comment]
Thanks @canvural, that was almost it! I've added a `TModelClass` template to ensure we can also pass subclasses of `Illuminate\Database\Eloquent\Model`. 

The tests succeed now.

**Comment by canvural** [comment]
Thank you.

**Comment by harmenjanssen** [comment]
Thanks for the merge!

**Pull request merged**